lib/external/math.dart

```diff
@@ -40,4 +50,128 @@ ASTNode funcRand(Runtime runtime, ASTNode self, List<ASTNode> args) {
   }
 
   throw const UnexpectedTypeException('The rand method only takes [String, double, int, and bool] argument types');
+}
+
+ASTNode funcSin(Runtime runtime, ASTNode self, List<ASTNode> args){
+  runtimeExpectArgs(args, [ASTType.AST_ANY]);
+  final radian = args[0];
+  
+  if(radian is IntNode || radian is DoubleNode){
+    final DoubleNode result = DoubleNode()..doubleVal = sin(radian is DoubleNode ? radian.doubleVal : radian.intVal);
+    return result;
+  }
+  
+  throw const UnexpectedTypeException('The sin method only accept [double, int] argument types');
+}
+
+ASTNode funcCos(Runtime runtime, ASTNode self, List<ASTNode> args){
```

Comment:
Be specific about the return type instead of just ast node give it double node!

Reply:
No, what he did is correct. Take a look at https://github.com/Calamity210/BirbLang/blob/master/lib/utils/ast/ast_node.dart#L6 @Andy-Python-Programmer

Reply:
Ah sorry for that @Calamity210